packages/inter-protocol/test/provisionPool.test.js

```diff
@@ -576,6 +576,7 @@ test('provisionPool publishes metricsOverride promptly', async t => {
   const {
     head: { value: initialMetrics },
   } = await E(metrics).subscribeAfter();
+  // FIXME fails when fakeVatAdmin.js fixed for non-promise root. Why?
```

Comment:
I found that adding a single `await null` after the `E(zoe).startInstance()` and before the `getMetrics()` call was sufficient to let the test pass.

I'm gonna guess there was a pre-existing race here, which was covered up by the extra `await` that the old fakeVatAdmin performed.

What exactly is the guarantee/contract provided by `E(zoe).startInstance()` ? Do we have any reason to believe that it has executed enough to let the metrics call see the desired value? Especially since, in this test environment, all of [ the test code, zoe, and the contract code ] are running in the same domain, without the run-until-the-microtask-queue-is-empty behavior we'd expect from three distinct real vats?

Reply:
I think your diagnosis is likely correct, in which case this is a racy bug. Could you file an issue?

Resolving here, but please add the issue link to this conversation when you have it. Thanks.

Reply:
When you file that bug, please also replace the `// FIXME` comment above with a link to that bug. Thanks.

Reply:
for reference, https://github.com/Agoric/agoric-sdk/issues/9598 , but I'll add the FIXME comment